code/game/objects/items/weapons/cosmetics.dm

```diff
@@ -339,3 +339,68 @@
 			..()
 	else
 		..()
+
+/obj/item/weapon/pocket_mirror //shamelessly copypasted from [mirror.dm]
+	name = "pocket mirror"
+	desc = "Mirror mirror on the wall, who's the most robust of them all? Touching the mirror will bring out Nanotrasen's state of the art hair modification system."
+	icon = 'icons/obj/items.dmi'
+	icon_state = "pocket_mirror"
+	flags = FPRINT
+	w_class = W_CLASS_TINY
+
+	var/shattered = 0
+
+/obj/item/weapon/pocket_mirror/attack_self(mob/user as mob)
+	if (shattered)
+		return
+
+	if (ishuman(user))
+		var/mob/living/carbon/human/H = user
+		if (isvampire(H))
+			if (!(VAMP_MATURE in H.mind.vampire.powers))
+				to_chat(H, "<span class='notice'>You don't see anything.</span>")
+				return
+
+		if (user.hallucinating())
+			switch(rand(1,100))
+				if (1 to 20)
+					to_chat(H, "<span class='sinister'>You look like [pick("a monster","a goliath","a catbeast","a ghost","a chicken","the mailman","a demon")]! Your heart skips a beat.</span>")
+					H.Knockdown(4)
+					return
+				if (21 to 40)
+					to_chat(H, "<span class='sinister'>There's [pick("somebody","a monster","a little girl","a zombie","a ghost","a catbeast","a demon")] standing behind you!</span>")
+					H.emote("scream",,, 1)
+					H.dir = turn(H.dir, 180)
+					return
+				if (41 to 50)
+					to_chat(H, "<span class='notice'>You don't see anything.</span>")
+					return
+
+		var/userloc = H.loc
+		//handle normal hair
+		var/list/species_hair = valid_sprite_accessories(hair_styles_list, null, (H.species.name || null))
+		//gender intentionally left null so speshul snowflakes can cross-hairdress
+		if (species_hair.len)
+			var/new_style = input(user, "Select a hair style", "Grooming")  as null|anything in species_hair
+			if (userloc != H.loc)
+				return	//no tele-grooming
+			if (new_style)
+				H.h_style = new_style
+				H.update_hair()
+
+/obj/item/weapon/pocket_mirror/proc/shatter()
+	if (shattered)
+		return
+	shattered = 1
+	icon_state = "pocket_mirror_broke"
+	playsound(src, "shatter", 70, 1)
+	desc = "Oh no, seven years of bad luck!"
+
+/obj/item/weapon/pocket_mirror/kick_act()
+	shatter()
+	..()
+
+/obj/item/weapon/pocket_mirror/throw_impact(atom/hit_atom)
+	..()
+	if (prob(25))
```

Comment:
`throw_impact()` is fucking stupid and gets called once for every object on the tile the thing stops on, so this is nigh guaranteed to pass if there are several things on the turf.
If that's how you want it to be, that's fine, but if you want it to be a flat 25% chance per throw, return after the `..()` if `hit_atom` is not a turf.